### PR TITLE
[core][state][dashboard][log] Fix subdirectory log getting

### PR DESCRIPTION
**Pull request by rickyyx**
## Why are these changes needed?
We are not able to get logs from a subdirectory from state API. This PR fixed it. 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: dashboard/modules/log/log_agent.py [hunk 1]
```diff
@@ -296,7 +296,7 @@ async def ListLogs(self, request, context):
             )
         log_files = []
         for p in path.glob(request.glob_filter):
-            log_files.append(p.name)
+            log_files.append(str(p.relative_to(path)))
```
Comment (rickyyx) [inline dashboard/modules/log/log_agent.py:299]: This is the main change. We don't include subdirectory parent directories before, so the list results are not streamable.

File: dashboard/modules/log/log_agent.py [hunk 2]
```diff
@@ -385,8 +385,8 @@ async def StreamLog(self, request, context):
         lines = request.lines if request.lines else 1000
         task_id = request.task_id if request.HasField("task_id") else None
 
-        filepath = f"{self._dashboard_agent.log_dir}/{request.log_file_name}"
-        if not os.path.isfile(filepath):
+        filepath = Path(f"{self._dashboard_agent.log_dir}") / f"{request.log_file_name}"
```
Comment (rkooo567) [inline dashboard/modules/log/log_agent.py:388]:
```suggestion
        filepath = Path(self._dashboard_agent.log_dir) / request.log_file_name
```

Can you delete a fstring here?

File: python/ray/experimental/state/state_manager.py [hunk 3]
```diff
@@ -440,11 +440,7 @@ async def stream_log(
...
-        await self._validate_stream(stream)
-        return stream
-
-    @staticmethod
-    async def _validate_stream(stream):
         metadata = await stream.initial_metadata()
         if metadata.get(log_consts.LOG_GRPC_ERROR) == log_consts.FILE_NOT_FOUND:
-            raise ValueError('File "{log_file_name}" not found on node {node_id}')
+            raise ValueError(f'File "{log_file_name}" not found on node {node_id}')
+        return stream
```
Comment (rkooo567) [inline python/ray/experimental/state/state_manager.py:446]: Does it have meaning to return here?

**Review by rkooo567 — APPROVED** [review]
NITs

File: python/ray/experimental/state/state_manager.py (hunk 3 above)
Comment (rickyyx) [reply python/ray/experimental/state/state_manager.py:446]: Yes? Because I removed the `_validate_stream` func. this is the same return as before.

Comment (rkooo567) [reply python/ray/experimental/state/state_manager.py:446]: oh yeah I was confused that this raise always happens...

**Comment by rkooo567** [comment]
Failures seem unreltaed

**Pull request merged**